src/primitives/triangle/mod.rs

```diff
@@ -144,39 +138,37 @@ impl ContainsPoint for Triangle {
 
 impl Dimensions for Triangle {
     fn bounding_box(&self) -> Rectangle {
-        let x_min = min(min(self.p1.x, self.p2.x), self.p3.x);
-        let y_min = min(min(self.p1.y, self.p2.y), self.p3.y);
+        let [p1, p2, p3] = self.vertices;
 
-        let x_max = max(max(self.p1.x, self.p2.x), self.p3.x);
-        let y_max = max(max(self.p1.y, self.p2.y), self.p3.y);
+        let x_min = min(min(p1.x, p2.x), p3.x);
+        let y_min = min(min(p1.y, p2.y), p3.y);
+
+        let x_max = max(max(p1.x, p2.x), p3.x);
+        let y_max = max(max(p1.y, p2.y), p3.y);
 
         Rectangle::with_corners(Point::new(x_min, y_min), Point::new(x_max, y_max))
     }
 }
 
 impl Triangle {
-    /// Create a new triangle with a given style
+    /// Create a new triangle with the given vertices.
     pub const fn new(p1: Point, p2: Point, p3: Point) -> Self {
-        Triangle { p1, p2, p3 }
+        Triangle {
+            vertices: [p1, p2, p3],
+        }
     }
 
-    /// Creates a new triangle from an array of points.
+    /// Creates a new triangle from the first three elements of a [`Point`] slice.
+    ///
+    /// # Panics
     ///
-    /// This supports both [`Point`]s, as well as anything that implements `Into<Point>` like
-    /// `(i32, i32)`.
+    /// This method will panic if the given slice is less than 3 items long.
```

Comment:
Perhaps we should panic for all length other than 3. This would match the behavior of `copy_from_slice`.